### DISCUSSION TITLE
Where did the "Connections" property go?

### DISCUSSION THREAD

**Discussion by tmontney** [sukona/Grapevine#231] [Q&A] [answered]
Upgraded an old build from 4.1.0 to 4.2.2. _Grapevine.Server.RestServer_ no longer has this property, to limit the max number of connections.

**Comment by scottoffen** [comment, marked as answer]
That property had been deprecated since by [4.1.0.0](https://github.com/scottoffen/grapevine-legacy/blob/v4.1.0.0/src/Grapevine/Server/ServerSettings.cs#L34) and was removed in 4.2.

Specifically, between [4.0.0.195](https://github.com/scottoffen/grapevine-legacy/tree/v4.0.0.195) and [4.0.0.252](https://github.com/scottoffen/grapevine-legacy/tree/v4.0.0.252) I switched from my previous, clumsy attempt to manage my own thread pool to using [`ThreadPool.QueueUserWorkItem`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadpool.queueuserworkitem?view=net-5.0), which does a much better job at it than I.  The `Connections` property was used in conjunction with the number of processors available to determine the size of the thread pool.

I allowed the property to perpetuate after this for backwards compatibility, but I added the obsolete attribute, as it no longer did anything.

**Reply by tmontney** [reply]
By that, you're saying I don't need to worry about this anymore? What if I wanted to limit the number of Connections? Is there a reason to do that still?

**Reply by scottoffen** [reply]
You don't need to worry about this anymore in Grapevine.

I can't think of a reason to continue limiting the number of connections. However, if that is a feature that you like to see added back in, I can move this discussion over to the ideas board, and we can figure out an approach.
